src/Bounds.cpp

```diff
@@ -892,6 +892,11 @@ class Bounds : public IRVisitor {
             // loop to check perfect nesting.
             interval = Interval(Call::make(Int(32), Call::buffer_get_min, op->args, Call::Extern),
                                 Call::make(Int(32), Call::buffer_get_max, op->args, Call::Extern));
+        } else if (op->is_intrinsic(Call::popcount) ||
+                   op->is_intrinsic(Call::count_leading_zeros) ||
+                   op->is_intrinsic(Call::count_trailing_zeros)) {
+            internal_assert(op->args.size() == 1);
+            interval = Interval(0, op->args[0].type().bits());
```

Comment:
Do these intrinsics always return Int(32)? If not you want Interval(make_zero(op->type), make_const(..., op->type))